### PR TITLE
Checkpoint review

### PR DESCRIPTION
**Pull request by andela-vkironde**
Kindly review this pull request

### CODE REVIEW

File: lib/Models/Song.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    self.artist = artist if artist
+    self.genre = genre if genre
+  end
+
+  def artist=(value)
+    @artist = value
+    @artist.add_song(self)
+  end
+
+  def genre=(value)
```
Comment (andela-hmasila) [inline lib/Models/Song.rb]: variable names should be self explanatory. `value` is too general

File: lib/Models/Song.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+class Song
+  extend Concerns::Findable
+  extend Concerns::Base
+  include Concerns::Saving
```
Comment (andela-hmasila) [inline lib/Models/Song.rb]: Use inheritance

File: README.md [hunk 3]
```diff
@@ -1,105 +1,73 @@
-# Ruby Music Library
+[![Build Status](https://travis-ci.org/andela-vkironde/checkpoint-1.svg?branch=develop)](https://travis-ci.org/andela-vkironde/checkpoint-1)
+#Music Library Application
```
Comment (andela-hmasila) [inline README.md]: Add a GPA badge

File: lib/Models/Artist.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Artist
+  extend Concerns::Findable
+  extend Concerns::Base
```
Comment (andela-hmasila) [inline lib/Models/Artist.rb]: Use inheritance

File: lib/Models/Genre.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+class Genre
+  extend Concerns::Findable
```
Comment (andela-hmasila) [inline lib/Models/Genre.rb]: Use inheritance

File: lib/Models/Artist.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  include Concerns::Saving
+  attr_accessor :name, :songs
+  @@all = []
+  def initialize(name)
```
Comment (andela-vkironde) [inline lib/Models/Artist.rb]: **Nitpick - Code Quality**
Add a new line before this line

File: README.md (hunk 3 above)
Comment (andela-vkironde) [inline README.md]: Done

File: lib/Models/Song.rb (hunk 1 above)
Comment (andela-vkironde) [inline lib/Models/Song.rb]: Done

File: lib/Models/Song.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  attr_reader :artist, :genre
+  @@all = []
+
+  def initialize(name, artist = nil, genre = nil)
+    @name = name
+    self.artist = artist if artist
+    self.genre = genre if genre
+  end
+
+  def artist=(value)
```
Comment (andela-vkironde) [inline lib/Models/Song.rb]: Done

File: lib/Models/Song.rb [hunk 8]
```diff
@@ -0,0 +1,36 @@
+class Song
+  extend Concerns::Findable
+  extend Concerns::Base
```
Comment (andela-vkironde) [inline lib/Models/Song.rb]: Done

File: lib/Models/Artist.rb (hunk 6 above)
Comment (andela-vkironde) [inline lib/Models/Artist.rb]: Done

**Pull request closed without merging**